ElementX/Sources/FlowCoordinators/RoomFlowCoordinator.swift

```diff
@@ -325,7 +325,19 @@ class RoomFlowCoordinator: FlowCoordinatorProtocol {
                 return .pollForm
             case (.pollForm, .dismissPollForm):
                 return .room
-            
+                
+            case (.room, .presentPinnedEventsTimeline):
+                return .pinnedEventsTimeline(source: .room)
+            case (.roomDetails(let isRoot), .presentPinnedEventsTimeline):
+                return .pinnedEventsTimeline(source: .details(isRoot: isRoot))
```

Comment:
Could we call these like so for some consistency?
```suggestion
                return .pinnedEventsTimeline(previousState: .roomDetails(isRoot: isRoot))
```